resource-managers/kubernetes/docker/src/main/dockerfiles/spark-base/Dockerfile

```diff
@@ -0,0 +1,47 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM openjdk:8-alpine
```

Comment:
The base image **openjdk:8-alpine** uses **musl libc** instead of **glibc**, so certain library (like Intel MKL for native BLAS support in mllib) cannot run on the container.  
There is also report on openjdk compatibility issue on alpine (https://github.com/apache-spark-on-k8s/spark/issues/326). 
Should we change to a base image that supports glibc?

Reply:
I wonder if there are any follow-ups of this particular issue (apache-spark-on-k8s#326).

Reply:
The docker files in this PR are for references only for now, and are by no means the official docker images moving forward. We will address concerns around the base image as part of SPARK-22647.

Reply:
@jason-dai Can you follow up on https://github.com/apache-spark-on-k8s/spark/issues/326 directly?

Reply:
I don't think it makes sense if the reference docker image does not work for common use cases because of the glibc issues (see discussions at https://github.com/apache-spark-on-k8s/spark/issues/559 and https://github.com/apache-spark-on-k8s/spark/issues/326). Now that we are in the process of merging the feature to Spark, I think we should fix these issues (or simply remove the reference image) before the merge.

Reply:
/cc @foxish @mccheah thoughts on finding and testing an alternative base image vs. removing it from this PR?

Reply:
When we send in the integration tests they will need to specify images so we're going to have to decide sooner or later.

I think using centos is fine, or fixing the base image to support glibc? Can't the Docker image be configured to use glibc despite using alpine?

Reply:
I think we can do what's done in https://github.com/apache-spark-on-k8s/spark/pull/550 by adding libc6-compat. From what I read, libc6 and glibc seem to be the same version of libc. See http://www.linux-m68k.org/faq/glibcinfo.html.

Reply:
Done in https://github.com/apache/spark/pull/19717/commits/67bc847e4f71bd6974735ac43090a0df266bc5f3.

Reply:
I'm in favor of using centos.  The compat layer makes me a bit nervous.  It may work fine but it smells like the kind of thing that could cause inscrutable binary incompatability bugs

Reply:
@yangw1234, can you confirm that libc6-compat fixed the issue you had?

Reply:
@foxish The libc6-compat did not fix our issue because libiomp5.so can not be loaded.

Reply:
@yangw1234 Adding `libc6-compat` on the spark-k8s base image works for me. I read and write parquet without issue. May be you had it only on the 'driver` image or made other changes that override the correct behavior (just random guess...)?